### PR TITLE
DEV 2147 show one time portal

### PR DESCRIPTION
**Pull request by x110dc**
#### What's this PR do?

- It allows one-time payments to be visible in the contributions portal. It does that by considering any Charge that doesn't have a linked Invoice as a one-time payment.

#### Why are we doing this? How does it help us?

- We want contributors to be able to view both one-time and recurring contributions.

#### How should this be manually tested? Please include detailed step-by-step instructions.

- Make a one-time donation here: https://billypenn-dev-2147.revengine-review.org/
- Request a magic link here: https://billypenn-dev-2147.revengine-review.org/contributor
- Follow the link in the email and confirm that the one-time contribution you just made is visible

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

- The Charge API is the older one and Stripe recommends using the Payment Intents API instead. This PR doesn't change from Charge -> Payment Intents but a future one probably should.
- The existing code determines the "next payment date" for recurring donations incorrectly. This PR doesn't fix that.

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2147

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

- Fix next payment date: https://news-revenue-hub.atlassian.net/browse/DEV-2192
- Use Payment Intents API instead of Charge API: https://news-revenue-hub.atlassian.net/browse/DEV-2193

### CODE REVIEW

File: apps/contributions/stripe_contributions_provider.py [hunk 1]
```diff
@@ -52,19 +48,22 @@ class StripeCharge:
...
+            return [{}]
         line_item = self.charge.invoice.lines.data
         if not line_item:
             line_item = [{}]
         return line_item[0]
 
     @property
     def interval(self):
+        if not self.charge.invoice:
+            # if there's no invoice then the it's a one-time payment
```
Comment (njharman) [inline apps/contributions/stripe_contributions_provider.py:65]: typo,  "then the it's", unneeded "the"

File: apps/contributions/stripe_contributions_provider.py [hunk 2]
```diff
@@ -52,19 +48,22 @@ class StripeCharge:
     """
```
Comment (njharman) [inline apps/contributions/stripe_contributions_provider.py:48]: Would like to see documented the rule "if no self.charge then this is a one-time payment.

**Review by njharman — APPROVED** [review]
changed code is 100% covered, 
looks good; approving with requested doc addition and comment typo correction.

File: apps/contributions/stripe_contributions_provider.py (hunk 1 above)
Comment (x110dc) [reply apps/contributions/stripe_contributions_provider.py:65]: fixed

File: apps/contributions/stripe_contributions_provider.py (hunk 2 above)
Comment (x110dc) [reply apps/contributions/stripe_contributions_provider.py:48]: Okay, noted.

**Pull request merged**